### DISCUSSION TITLE
Allow string / number primitive values on many to one filtering in GraphQL

### DISCUSSION THREAD

**Discussion by braco** [directus/directus#24867] [Feature Requests]
@rijkvanzanten's note: The GraphQL spec doesn't support input union types, so Directus defaults to the nested filter input type for m2o fields. However, primitive filters should be allowed on those fields as well

---

### Describe the Bug

Using the graphql interface, how do you find all records that _don't have a file_ in a certain field?

Queries like  { fileField: { _empty: true } } yield 

"message": "Field "_empty" is not defined by type "directus_files_filter".",

Queries like  { fileField: { id { _empty: true } } } yield 

message: '"uuid" field type does not contain the "_empty" filter operator',

I found a temporary workaround by checking if the `fileField { filesize }` field is null, but this seems hacky?

```
              fileField: {
                filesize: {
                  _null: true
                }
              }
```

### To Reproduce

see above

### Directus Version

v10.2.0

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by rijkvanzanten** [comment]
`_empty` checks against an empty string rather than `null`. GraphQL doesn't support union input types, so it currently defaults to treating it relationally. Your example is therefore very close; I'd personally check `id` for `null` instead of filesize, but the end result is the same.

I'll repurpose this issue to see if we can somehow "fake" a union input type, as ideally you should just be able to use `{ filter: { fileField: { _null: true } } }`!
